src/lib/rosetta_models/enums.ml

```diff
@@ -2,10 +2,10 @@ type curvetype = (* pallas *) string [@@deriving yojson, show]
 
 type signaturetype = (* schnorr_poseidon *) string [@@deriving yojson, show]
 
-type coinaction = () [@@deriving yojson, show, eq]
+type coinaction = unit [@@deriving yojson, show, eq]
```

Comment:
These are supposed to be distinct constructors so that they don't accidentally alias. Could you replace these with named constructors?

Reply:
using constructors like `Coin_action` ... seems to work.